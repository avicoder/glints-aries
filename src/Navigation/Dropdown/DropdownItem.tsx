import * as React from 'react';

import { DropdownItemWrapper } from './DropdownStyle';

const DropdownItem: React.FunctionComponent<Props> = props => {
  const { children, className, ...defaultProps } = props;

  return (
    <DropdownItemWrapper className={className} {...defaultProps}>
      {children}
    </DropdownItemWrapper>
  );
};

export interface Props
  extends React.ComponentPropsWithoutRef<typeof DropdownItemWrapper> {
  value?: any;
  onClick?: (e: React.MouseEvent<HTMLLIElement, MouseEvent>) => void;
  children: React.ReactNode;
}

export default DropdownItem;
