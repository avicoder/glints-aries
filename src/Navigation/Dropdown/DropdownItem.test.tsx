import * as React from 'react';

import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

import DropdownItem from './DropdownItem';

describe('<DropdownItem /> prop className', () => {
  const matchSnapshot = (className: string) => {
    test(`should match snapshot when class name ${className} is passed`, () => {
      const { asFragment } = render(
        <DropdownItem className={className}>Product Manager</DropdownItem>
      );

      expect(asFragment()).toMatchSnapshot();
    });
  };

  ['test', undefined].forEach(matchSnapshot);
});
