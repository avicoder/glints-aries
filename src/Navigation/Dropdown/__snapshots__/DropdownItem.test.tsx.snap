// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownItem /> prop className should match snapshot when class name test is passed 1`] = `
<DocumentFragment>
  <li
    class="DropdownStyle__DropdownItemWrapper-sc-1sc3c6h-4 bnpgGr test"
  >
    Product Manager
  </li>
</DocumentFragment>
`;

exports[`<DropdownItem /> prop className should match snapshot when class name undefined is passed 1`] = `
<DocumentFragment>
  <li
    class="DropdownStyle__DropdownItemWrapper-sc-1sc3c6h-4 bnpgGr"
  >
    Product Manager
  </li>
</DocumentFragment>
`;
